xml_converter/src/attribute/color.cpp

```diff
@@ -28,27 +38,48 @@ Color parse_color(rapidxml::xml_attribute<>* input, vector<XMLError*>*) {
 // Converts a Color into a stringy value so it can be saved to xml.
 ////////////////////////////////////////////////////////////////////////////////
 string stringify_color(Color attribute_value) {
-    return attribute_value.hex;
+    return "#" + attribute_value.hex;
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 // to_proto_color
 //
 // Converts a Color into a proto message
+// TODO: Find a way to transfer alpha when color hex is not set
 ////////////////////////////////////////////////////////////////////////////////
-waypoint::Color* to_proto_color(Color attribute_value) {
-    waypoint::Color* color = new waypoint::Color();
-    color->set_hex(attribute_value.hex);
-    return color;
+waypoint::RGBA* to_proto_color(Color attribute_value, float alpha) {
+    waypoint::RGBA* rgba = new waypoint::RGBA();
+    string hex = attribute_value.hex;
+    int r = stoi(hex.substr(0, 2), 0, 16);
+    int g = stoi(hex.substr(2, 2), 0, 16);
+    int b = stoi(hex.substr(4, 2), 0, 16);
+
+    int a;
+    if (alpha == 1.0) {
+        a = 255;
+    }
+    else {
+        a = static_cast<int>(alpha * 256);
+    }
+
+    uint32_t rgba_int = ((r & 0xff) << 24) + ((g & 0xff) << 16) + ((b & 0xff) << 8) + (a & 0xff);
+
+    rgba->set_rgba(rgba_int);
+    return rgba;
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 // from_proto_color
 //
 // Converts a proto message into a Color
 ////////////////////////////////////////////////////////////////////////////////
-Color from_proto_color(waypoint::Color attribute_value) {
+Color from_proto_color(waypoint::RGBA attribute_value) {
     Color color;
-    color.hex = attribute_value.hex();
+    std::stringstream stream;
+    stream << std::hex << attribute_value.rgba();
+    std::string rgba = stream.str();
+
+    color.hex = rgba.substr(0, 6);
+    color.alpha = std::stof(rgba.substr(6, 2));
```

Comment:
Is this right? Does std::stof() convert from a 2 digit hex value equivalent to (0-255) into a 0.0 - 1.0 float value automatically?